### PR TITLE
♻ refactor(presets): refactor Modal and Drawer

### PR DESCRIPTION
**Pull request by capdiem**
- Remove preset component Actions
- Add the base component ModalBase, the new Modal and Drawer are inherit from it
- Refactor preset component Modal and Drawer
  - Delete preset component FormModal, and add FormModel parameter to support the Form feature
  - Add OnDelete parameter
  - Add OkProps, CancelProps, DeleteProps parameters to set action style
  - Replace ActionsContent with OkContent, CancelContent and DeleteContent

### CODE REVIEW

File: src/Doc/MASA.Blazor.Doc/Demos/Components/Presets/children/Modal/contents/CustomActions.razor [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        OnCancel="OnCancel"
+        OnDelete="OnDelete">
+    <ActivatorContent>
+        <MButton @attributes="@context.Attrs">Activator</MButton>
+    </ActivatorContent>
+    <ChildContent>
+        Content
+    </ChildContent>
+    <DeleteContent Context="delete">
+        <MButton Icon Color="error" OnClick="delete.click" Disabled="delete.loading">
```
Comment (doddgu) [inline src/Doc/MASA.Blazor.Doc/Demos/Components/Presets/children/Modal/contents/CustomActions.razor:14]: click -> Click

File: src/Doc/Masa.Blazor.Doc/Demos/Components/Presets/children/Modal/props/FormModel.razor [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                        HideDetails="@("auto")" />
+        </MCol>
+        <MCol Cols="12">
+            <MRadioGroup @bind-Value="userModel.Gender"
+                         Label="Gender:"
+                         Row
+                         Dense
+                         Style="margin-top: 0"
+                         HideDetails="@("auto")">
+                <MRadio Label="男" Value="0"></MRadio>
```
Comment (doddgu) [inline src/Doc/Masa.Blazor.Doc/Demos/Components/Presets/children/Modal/props/FormModel.razor:42]: 男 -> Male

File: src/Masa.Blazor/Presets/Modal/ModalBase.razor.cs [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+        protected ModalButtonProps ComputedDeleteButtonProps { get; set; }
+
+        public override async Task SetParametersAsync(ParameterView parameters)
+        {
+            AutoScrollToTop = true;
+
+            await base.SetParametersAsync(parameters);
+
+            OkText ??= "确定";
```
Comment (doddgu) [inline src/Masa.Blazor/Presets/Modal/ModalBase.razor.cs:152]: Support I18n

File: src/Masa.Blazor/Presets/Modal/ModalBase.razor.cs [hunk 4]
```diff
@@ -149,9 +149,9 @@ public override async Task SetParametersAsync(ParameterView parameters)
 
             await base.SetParametersAsync(parameters);
 
-            OkText ??= "确定";
-            CancelText ??= "取消";
-            DeleteText ??= "删除";
+            OkText ??= "OK";
```
Comment (doddgu) [inline src/Masa.Blazor/Presets/Modal/ModalBase.razor.cs:152]: Ok -> Submit

Comment (capdiem) [reply src/Masa.Blazor/Presets/Modal/ModalBase.razor.cs]: Ok -> Save

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request closed without merging**